RSS Reader/RSS Reader/Topics list/Presenter/RSSRTopicsListPresenter.m

```diff
@@ -0,0 +1,85 @@
+//
+//  RSSRTopicsListPresenter.m
+//  RSS Reader
+//
+//  Created by Lina Loyko on 11/18/20.
+//
+
+#import <UIKit/UIKit.h>
+#import "RSSRTopicsListPresenter.h"
+#import "RSSRTopic.h"
+#import "RSSRNetworkService.h"
+#import "RSSRXMLParser.h"
+#import "UIViewController+AlertPresentable.h"
+
+@interface RSSRTopicsListPresenter ()
+
+@property (nonatomic, assign) id<RSSFeedView, AlertPresentable> feedView;
+@property (nonatomic, retain) RSSRNetworkService *networkService;
+@property (nonatomic, retain) RSSRXMLParser *parser;
+@property (nonatomic, retain) NSArray<RSSRTopic *> *dataSource;
+
+@end
+
+@implementation RSSRTopicsListPresenter
+
+static NSString * const baseURLString = @"http://news.tut.by/rss/index.rss";
+
+- (instancetype)initWithService:(RSSRNetworkService *)service parser:(RSSRXMLParser *)parser {
+    self = [super init];
+    if (self) {
+        _networkService = [service retain];
+        _parser = [parser retain];
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [_networkService release];
+    [_parser release];
+    [_dataSource release];
+    [super dealloc];
+}
+
+- (void)parseTopicsData:(NSData *)data {
+    __block typeof(self) weakSelf = self;
+    [self.parser parseTopics:data completion:^(NSMutableArray<RSSRTopic *> *topics, NSError *error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            if (error) {
+                [weakSelf showError:error];
+            } else {
+                weakSelf.dataSource = topics;
+                [weakSelf.feedView reloadData];
+            }
+        });
+    }];
+}
+
+- (void)showError:(NSError *)error {
+    [self.feedView showAlertWithTitle:@"Error" message:error.localizedDescription];
+}
+
+
+#pragma mark - RSSRFeedPresenter Protocol
+
+- (void)loadTopics {
+    __block typeof(self) weakSelf = self;
+    [self.networkService loadDataFromURL:[NSURL URLWithString:baseURLString]
+                         completion:^(NSData *data, NSError *error) {
+        if (error) {
+            [weakSelf showError:error];
```

Comment:
why not just call '[self.feedView showAlertWithTitle:@"Error" message:error.localizedDescription];' here?

Reply:
I call this method in two places. And if i want to change the way I show error, I’ll only need to do it once